app/helpers/schedules_helper.rb

```diff
@@ -0,0 +1,7 @@
+module SchedulesHelper
+  def weeks_in_location(location)
+    date_1 = location&.start_date&.to_time
+    date_2 = location&.end_date&.to_time
+    (date_2 - date_1).seconds.in_weeks.to_i.abs if date_1.present? && date_2.present?
```

Comment:
It might be more helpful if weeks_in_location() (or another method) generated an array of dates, each one being the start of a week in the program year. Something like:

```
   weeks = []
   week = location.start_date
   last_week = location.end_date - 5.days
   while week != last_week
      weeks << week
      week += 7.days
   end

   weeks
```

Eventually each week slot in the program year will be assigned an event.

Reply:
@brentkearney - this method actually giving us number of weeks between start_date and end_date of a particular location as mentioned in ticket and below screenshot:
https://trello.com/c/tFGjsCb1/685-add-schedule-events-section-with-program-parameters
file:///home/hassan/Pictures/params%20(1).png